server/pg/sql_analyzer_velox.cpp

```diff
@@ -2658,19 +2683,37 @@ SqlAnalyzer::TableAliasAndColumnNames SqlAnalyzer::ProcessTableColumns(
 State SqlAnalyzer::ProcessTable(State* parent, std::string_view schema_name,
                                 std::string_view table_name,
                                 const Objects::ObjectData& object,
-                                const RangeVar* node) {
+                                const RangeVar* node, bool load_implicit_pk) {
   const auto& table = basics::downCast<catalog::Table>(*object.object);
-  const auto& type = *table.RowType();
-  auto [table_alias, column_names] =
-    ProcessTableColumns(parent, node, table.RowType());
+  auto type = table.RowType();
+
+  auto [table_alias, column_names] = ProcessTableColumns(parent, node, type);
 
   object.EnsureTable();
 
+  if (load_implicit_pk) {
+    SDB_ASSERT(object.object);
+    if (table.PKColumns().empty()) {
```

Comment:
```suggestion
  if (load_implicit_pk && table.PKColumns().empty()) {
```
this assert is always true because code above, you can move it above if you want